### PR TITLE
ECO-3055: Create xml parsing logic

### PR DESCRIPTION
**Pull request by aleksey-kotsuba**

### CODE REVIEW

File: src/SprykerEco/Client/Heidelpay/HeidelpayClient.php [hunk 1]
```diff
@@ -211,6 +212,24 @@ public function getCreditCardPaymentOptions(QuoteTransfer $quoteTransfer): Heide
             ->getCreditCardPaymentOptions($quoteTransfer);
     }
 
+    /**
```
Comment (vhrychenko) [inline src/SprykerEco/Client/Heidelpay/HeidelpayClient.php]: inheritdoc?

Comment (aleksey-kotsuba) [reply src/SprykerEco/Client/Heidelpay/HeidelpayClient.php]: You are right.

File: src/SprykerEco/Zed/Heidelpay/Business/HeidelpayBusinessFactory.php [hunk 2]
```diff
@@ -699,73 +705,100 @@ public function createAesEncrypter(): EncrypterInterface
...
+            ->createAdapterFactory()
+            ->createCreditCardPaymentMethodAdapter();
     }
 
     /**
-     * @return \SprykerEco\Zed\Heidelpay\Dependency\Facade\HeidelpayToCurrencyFacadeInterface
+     * @return \SprykerEco\Zed\Heidelpay\Business\Basket\BasketCreatorInterface
      */
-    public function getCurrencyFacade(): HeidelpayToCurrencyFacadeInterface
+    public function createBasketHanlder(): BasketCreatorInterface
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/HeidelpayBusinessFactory.php]: mb `createBasketCreator`?

File: src/SprykerEco/Zed/Heidelpay/Business/Processor/Notification/Expander/NotificationExpander.php [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+     * @param \Generated\Shared\Transfer\HeidelpayNotificationTransfer $notificationTransfer
+     * @param array $transactionData
+     *
+     * @return \Generated\Shared\Transfer\HeidelpayNotificationTransfer
+     */
+    protected function addTransactionData(
+        HeidelpayNotificationTransfer $notificationTransfer,
+        array $transactionData
+    ): HeidelpayNotificationTransfer {
+        $notificationTransfer->setTransactionSource($transactionData['source']);
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Processor/Notification/Expander/NotificationExpander.php]: could this be constants?

File: src/SprykerEco/Zed/Heidelpay/Business/Processor/Notification/Converter/NotificationXmlConverter.php [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+     *
+     * @throws \SprykerEco\Zed\Heidelpay\Business\Exception\TransactionElementMissingException
+     *
+     * @return \SimpleXMLElement
+     */
+    protected function loadXml(string $xml): SimpleXMLElement
+    {
+        $xmlProcess = new SimpleXMLElement($xml);
+        if (!property_exists($xmlProcess, static::TRANSACTION_ELEMENT)) {
+            throw new TransactionElementMissingException();
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/Business/Processor/Notification/Converter/NotificationXmlConverter.php]: We might want to have a descriptive error message here.

File: src/SprykerEco/Zed/Heidelpay/HeidelpayDependencyProvider.php [hunk 5]
```diff
@@ -25,30 +25,103 @@ class HeidelpayDependencyProvider extends AbstractBundleDependencyProvider
 
     public const SERVICE_UTIL_ENCODING = 'util encoding service';
 
+    public const PLUGIN_HEIDELPAY_NOTIFICATION_EXPANDER = 'PLUGIN_HEIDELPAY_NOTIFICATION_EXPANDER';
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/HeidelpayDependencyProvider.php]: not a big blocker, but PLUGIN part should be plural

File: src/SprykerEco/Zed/Heidelpay/HeidelpayDependencyProvider.php [hunk 6]
```diff
@@ -25,30 +25,103 @@ class HeidelpayDependencyProvider extends AbstractBundleDependencyProvider
...
+    }
+
+    /**
+     * @param \Spryker\Zed\Kernel\Container $container
+     *
+     * @return \Spryker\Zed\Kernel\Container
+     */
+    protected function addSalesFacade(Container $container): Container
+    {
+        $container[static::FACADE_SALES] = function (Container $container) {
```
Comment (gechetspr) [inline src/SprykerEco/Zed/Heidelpay/HeidelpayDependencyProvider.php]: Could you use new container syntax? Also don't forget to raise `kernel` version dependency for it

**Pull request merged**